x-pack/plugins/actions/server/builtin_action_types/lib/connector_token_client.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ConnectorToken } from '../../types';
+import { EncryptedSavedObjectsClient } from '../../../../encrypted_saved_objects/server';
+import { SavedObjectsClientContract, SavedObjectsUtils } from '../../../../../../src/core/server';
+import { CONNECTOR_TOKEN_SAVED_OBJECT_TYPE } from '../../constants/saved_objects';
+
+// We are expecting max possible token types for the single connector to be not bigger then 2 - for access and refresh tokens
+export const MAX_TOKENS_RETURNED = 2;
+
+interface ConstructorOptions {
+  encryptedSavedObjectsClient: EncryptedSavedObjectsClient;
+  savedObjectsClient: SavedObjectsClientContract;
+}
+
+interface CreateOptions {
+  connectorId: string;
+  token: string;
+  expiresIn: string;
+  tokenType?: string;
+}
+
+export interface UpdateOptions {
+  id: string;
+  token: string;
+  expiresIn: string;
+  tokenType?: string;
+}
+
+export class ConnectorTokenClient {
+  private readonly savedObjectsClient: SavedObjectsClientContract;
```

Comment:
nit: should we call this `unsecuredSavedObjectsClient` to make clear it's unsecured?